src/parser/bison_parser.y

```diff
@@ -941,16 +946,28 @@ table_name:
 	;
 
 
+table_alias:
+		alias
+	|	AS IDENTIFIER '(' ident_commalist ')' { $$ = new Alias($2, $4); }
+	;
+
+
+opt_table_alias:
```

Comment:
still a big lack of understanding for bison on my side, but why do you have two separate rules opt_table_alias AND table_alias? Wouldn't opt_table_alias be sufficient?

Reply:
Technically one could merge `opt_x` and `x` rules, I do not know about a technical requirement. I would stick with separate clauses for the following reasons: 

1. The grammar rules are like building blocks. Possibly detailed control of mandatory and optional aliases is desired at a later stage. 
3. Inline with the rest of the grammar definitions.
4. It's clunky to nest optional clauses inside optional clauses. For instance, expressing `opt_table_alias` in one rule would result in the following:

        opt_table_alias:
    	    	opt_alias
    	    |	AS IDENTIFIER '(' ident_commalist ')' { $$ = new Alias($2, $4); }
    	    ;

    It does not seem to be as straightforward and obvious as the current proposal.

Reply:
Yepp, that makes sense to me.